cpp/tests/io/csv_test.cpp

```diff
@@ -255,6 +255,50 @@ void write_csv_helper(std::string const& filename,
   cudf_io::write_csv(writer_options);
 }
 
+template <typename T>
+std::vector<std::string> append_zeros(const std::vector<T>& input,
+                                      int zero_count         = 0,
+                                      bool add_positive_sign = false)
+{
+  std::vector<std::string> output(input.size());
+  std::transform(input.begin(), input.end(), output.begin(), [=](const T& num) {
+    auto str         = std::to_string(num);
```

Comment:
You can transform the input vector into std::vector<std::string> and then call the specialization, instead of duplicating the logic in this transform.